wfl/calculators/vasp.py

```diff
@@ -4,185 +4,216 @@
 """
 
 import os
-import pathlib
+from pathlib import Path
 import tempfile
-import warnings
+import shutil
+
 import json
+from copy import deepcopy
 
 import numpy as np
 
 from ase.atoms import Atoms
-from ase.calculators.vasp.vasp import Vasp
+from ase.calculators.calculator import all_changes
+import ase.calculators.vasp.vasp
 
-from .utils import clean_rundir, handle_nonperiodic, save_results
-from ..utils.misc import atoms_to_list
+from .utils import clean_rundir, handle_nonperiodic
 
 # NOMAD compatible, see https://nomad-lab.eu/prod/rae/gui/uploads
-__default_keep_files = ["POSCAR", "INCAR", "KPOINTS", "OUTCAR", "vasprun.xml", "vasp.out"]
-__default_properties = ["energy", "forces", "stress"]
-
-
-def evaluate_autopara_wrappable(
-    atoms,
-    workdir_root=None,
-    dir_prefix="run_VASP_",
-    calculator_command=None,
-    calculator_kwargs=None,
-    output_prefix="VASP_",
-    properties=None,
-    keep_files="default",
-):
-    """Evaluate a configuration with VASP
+_default_keep_files = ["POSCAR", "INCAR", "KPOINTS", "OUTCAR", "vasprun.xml", "vasp.out"]
+_default_properties = ["energy", "forces", "stress"]
+
+_vasp_kwargs_def = {
```

Comment:
I think this is fixed now. I had to do `pp` separately, since `Vasp` only uses the `default_parameters` class attributes for things that are INCAR tags, not ASE calculator infrastructure kwargs.

Reply:
> Hmm. Do we know that Vasp would use that class attribute in the same way as CP2K?
Yes, I think the default parameter mechanism comes from ASE, not CP2K. 

The change looks ok, thanks. 

> Actually, a more fundamental questions is whether we think that just on general principles we should, as much as possible, stick with ASE's defaults, even if we don't love them, and not even override these?

I would argue for sticking to ASE's defaults. I'd expect that the defaults are only used to quickly get going and test the calculator, no-one would routinely rely on the defaults (unless that person has coded up the defaults). In which case having one and not two sets of defaults would mean fewer places for confusion.

Reply:
OK, I generally agree.  The one exception is `ENCUT`.  I refuse to use VASP's default of max(POTCAR), because it's a terrible idea, especially for `wfl`. It means that if you have different compositions in different configs you'll get inconsistent results.  That's also sort of true for k-points, but maybe that's a sufficiently difficult thing to make consistent that we can't impose a sensible default (Although perhaps we should require either kspacing or kpoints).  But for ENCUT, that's something I want the user to have to pick.  I'll go over all the other non-standard defaults and see if there's anything else I feel strongly about, otherwise I'll remove them.

Reply:
sounds good. I guess I meant that I would rather stick with ASE's defaults for somewhat subjectively arbitrary choices (e.g. one favourite functional over the other), but I'm up for it if it generally makes the behaviour clearer or more sensible.

Reply:
I'm going to continue to require ENCUT explicitly, and I think I'll also switch the default ISMEAR to 0, because the regular default of 1 is actively wrong for insulators. I'm doing a bit of test, but probably everything else can be dropped. I do want to make sure that it doesn't do something silly about ISIF which leads to incorrect stresses.

Reply:
OK - I'm down to overriding `ismear` to 0 (because 1 is wrong for insulators), `lwave` and `lcharg` to False (because they take lots of space and some I/O time, and they're mostly useful for restarting calculations), and requiring an explicit `encut`. If you think that's a reasonable compromise, can you resolve this conversation?